### PR TITLE
Fix http_skip_not_found_url with compressed files

### PR DESCRIPTION
**Pull request by al13n321**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
 * Fix http_skip_not_found_url_for_globs not working for compressed files.

The behavior for skipping not-found URLs for globs was kind of inconsistent, in part broken by me in https://github.com/ClickHouse/ClickHouse/pull/47964

It skipped files by pretending that they're empty. This didn't work when the file is compressed - the decompression would error out trying to decompress an empty string.

Fixed by explicitly skipping the URL if the initial HTTP request failed, without pretending that the file is empty.

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
This is an automated comment for commit bacdbb307ac74d7f34a7d91390801b66386fadb1 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49541/bacdbb307ac74d7f34a7d91390801b66386fadb1/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/IO/ReadWriteBufferFromHTTP.cpp [hunk 1]
```diff
@@ -166,29 +166,23 @@ std::istream * ReadWriteBufferFromHTTPBase<UpdatableSessionPtr>::callImpl(
...
 size_t ReadWriteBufferFromHTTPBase<UpdatableSessionPtr>::getFileSize()
 {
-    if (!file_info)
-        file_info = getFileInfo();
-
-    if (file_info->file_size)
-        return *file_info->file_size;
+    auto r = getFileInfo().file_size;
+    if (r)
+        return *r;
```
Comment (kssenii) [inline src/IO/ReadWriteBufferFromHTTP.cpp:171]: what is `r`? May be use a bit more explanatory name?)

File: src/Storages/StorageURL.cpp [hunk 2]
```diff
@@ -272,7 +278,10 @@ namespace
 
                     auto current_uri = uri_info->uri_list_to_read[current_uri_pos];
 
-                    initialize(current_uri);
+                    auto r = initialize(current_uri);
```
Comment (kssenii) [inline src/Storages/StorageURL.cpp:281]: same for `r`

**Review by kssenii — APPROVED** [review]

**Comment by al13n321** [comment]
(It's already fixed)

**Pull request closed without merging**